### PR TITLE
Add new `vue/require-prop-comment` rule

### LINKED ISSUE #1993: require-prop-comment
**Please describe what the rule should do:**

require prop should have a comment

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[X ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
      <template>
        <div>1</div>
      </template>
      <script setup>
      const props = defineProps({
        a: Number
      })
      </script>
```
```vue
      <template>
        <div>1</div>
      </template>
      <script>
      import { defineComponent } from '@vue/composition-api'

      export default defineComponent({
        props: {
          /**
           * a comment
           */
          a: Number,
          b: Number,
          // c
          // comment
          c: Number
        },
        setup() {}
      })
      </script>
```

**Additional context**

### PR DESCRIPTION
**Pull request by czb3279338858**
Fixes #1993

### CODE REVIEW

**Comment by czb3279338858** [comment]
npm run lint,Changed the description of many files，I'm not sure this is correct, but I still submitted the code

**Comment by czb3279338858** [comment]
1 failing：The file causing the error is not modified by me. Should I repair it before submitting

File: docs/rules/README.md [hunk 1]
```diff
@@ -276,42 +277,42 @@ The following rules extend the rules provided by ESLint itself and apply them to
 
 | Rule ID | Description |    |    |
 |:--------|:------------|:--:|:--:|
-| [vue/array-bracket-newline](./array-bracket-newline.md) | enforce linebreaks after opening and before closing array brackets in `<template>` | :wrench: | :lipstick: |
```
Comment (FloEdelmann) [inline docs/rules/README.md:279]: `git pull origin master && npm run update` should probably fix these unrelated changes.

File: lib/rules/require-prop-comment.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    /**
+     * @param {import('../utils').ComponentProp[]} props
+     */
+    function verifyProps(props) {
+      for (const prop of props) {
+        if (!prop.propName) continue
+
+        const beforeComments = sourceCode.getCommentsBefore(prop.node)
+        let message = ''
```
Comment (FloEdelmann) [inline lib/rules/require-prop-comment.js:91]: Using an empty string is a bit unclean. Better explicitly allow `undefined` and change the functions to `return undefined` instead of `return ''`.

```suggestion
        /** @type {string|undefined} */
        let message
```

File: docs/rules/require-prop-comment.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+This rule enforces that every prop has a comment that documents it.
+
+<eslint-code-block :rules="{'vue/require-prop-comment': ['error']}">
+
+```vue
+<!-- ✓ GOOD -->
+<template>
+  <div>1</div>
+</template>
```
Comment (FloEdelmann) [inline docs/rules/require-prop-comment.md:23]: Not needed for this example:

```suggestion
```

File: docs/rules/require-prop-comment.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+</eslint-code-block>
+
+<eslint-code-block :rules="{'vue/require-prop-comment': ['error']}">
+
+```vue
+<!-- ✗ BAD -->
+<template>
+  <div>1</div>
+</template>
```
Comment (FloEdelmann) [inline docs/rules/require-prop-comment.md:51]: Again:

```suggestion
```

File: docs/rules/require-prop-comment.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+const props = defineProps({
+  /*
+   * a comment
+   */
+  /*
+   * a other comment
+   */
+  a: Number,
+})
+</script>
```
Comment (FloEdelmann) [inline docs/rules/require-prop-comment.md:100]: Why is this bad? There is a block comment before the `a` prop.

File: tests/lib/rules/require-prop-comment.js [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+  }
+})
+
+tester.run('require-prop-comment', rule, {
+  valid: [
+    {
+      code: `
+      <template>
+        <div>1</div>
+      </template>
```
Comment (FloEdelmann) [inline tests/lib/rules/require-prop-comment.js:24]: Please remove the template from all tests, it is superfluous in this rule.

**Review by FloEdelmann — CHANGES_REQUESTED** [review]

File: docs/rules/README.md (hunk 1 above)
Comment (czb3279338858) [reply docs/rules/README.md]: complete

File: docs/rules/require-prop-comment.md (hunk 4 above)
Comment (czb3279338858) [reply docs/rules/require-prop-comment.md]: This test is to check block，default is JSDoc

File: docs/rules/require-prop-comment.md (hunk 5 above)
Comment (czb3279338858) [reply docs/rules/require-prop-comment.md]: block and JSDoc
1、Multiple lines are supported, and it is unnecessary to have multiple lines at the same time
2、The document generator may have difficulty recognizing which is what it needs

File: tests/lib/rules/require-prop-comment.js (hunk 6 above)
Comment (czb3279338858) [reply tests/lib/rules/require-prop-comment.js]: complete

**Comment by czb3279338858** [comment]
Where there is dispute, should I initiate a new merger request? @FloEdelmann

**Comment by FloEdelmann** [comment]
You don't need to open a new PR, you can instead push new commits to this branch. Also, rebasing and force-pushing is not required, we will squash the whole branch into one commit on the `master` branch when merging the PR.

File: docs/rules/require-prop-comment.md (hunk 3 above)
Comment (czb3279338858) [reply docs/rules/require-prop-comment.md]: I misunderstood your meaning. The template has been deleted

File: docs/rules/require-prop-comment.md (hunk 5 above)
Comment (FloEdelmann) [reply docs/rules/require-prop-comment.md]: I'd say this is an edge case that doesn't need to be handled in this rule. If the rule logic can be simplified by looking only if there is *a* comment for a prop (and not if there is *exactly one* comment), then we should rather do that.

File: docs/rules/require-prop-comment.md (hunk 4 above)
Comment (czb3279338858) [reply docs/rules/require-prop-comment.md]: Completed

File: docs/rules/require-prop-comment.md (hunk 5 above)
Comment (czb3279338858) [reply docs/rules/require-prop-comment.md]: complete

**Comment by czb3279338858** [comment]
@FloEdelmann I push new commits.
